src/csharp/Microsoft.Spark/Utils/UdfUtils.cs

```diff
@@ -183,22 +182,23 @@ internal static JvmObjectReference CreatePythonFunction(IJvmBridge jvm, byte[] c
         private static IJvmObjectReferenceProvider CreateEnvVarsForPythonFunction(IJvmBridge jvm)
         {
             var environmentVars = new Hashtable(jvm);
-            string assemblySearchPath = string.Join(",",
-                new[]
-                {
-                    Environment.GetEnvironmentVariable(
-                        AssemblySearchPathResolver.AssemblySearchPathsEnvVarName),
-                    SparkFiles.GetRootDirectory()
-                }.Where(s => !string.IsNullOrWhiteSpace(s)));
-
+            string assemblySearchPath = Environment.GetEnvironmentVariable(
+                AssemblySearchPathResolver.AssemblySearchPathsEnvVarName);
             if (!string.IsNullOrEmpty(assemblySearchPath))
             {
                 environmentVars.Put(
                     AssemblySearchPathResolver.AssemblySearchPathsEnvVarName,
                     assemblySearchPath);
             }
-            // DOTNET_WORKER_SPARK_VERSION is used to handle different versions of Spark on the worker.
-            environmentVars.Put("DOTNET_WORKER_SPARK_VERSION", SparkEnvironment.SparkVersion.ToString());
+            // DOTNET_WORKER_SPARK_VERSION is used to handle different versions
+            // of Spark on the worker.
+            environmentVars.Put(
+                "DOTNET_WORKER_SPARK_VERSION",
+                SparkEnvironment.SparkVersion.ToString());
+
+            string replMode =
+                EnvironmentUtils.GetEnvironmentVariableAsBool("DOTNET_SPARK_REPL_MODE").ToString();
+            environmentVars.Put("DOTNET_SPARK_REPL_MODE", replMode);
```

Comment:
nit: do we always to set this or only when it's running a repl?  Also, I suggested this name, but mode may not be a good name. Maybe `DOTNET_SPARK_RUNNING_REPL` or a name that conveys a boolean may be better?

Reply:
Only need to set it if it's running the REPL.  Modified code a bit and renamed env variable.

Reply:
Hmm, where do you think we should set this env variable?  Should we set it from the extension or whatever is going to start the dotnet-interactive session (livy) ?

Reply:
good question. I was thinking whoever is starting the dotnet process, but I guess extension can still set it as well to be sure?

Reply:
I think whoever is starting up the `DotnetRunner` (or `DotnetBackend`).